### ISSUE TITLE
Why NUnitLite

### ISSUE THREAD

**Issue by manfred-brands** [nunit/nunit#4445]
The build script for nunit tests using `NUnitLite`.

When building, we use `Visual Studio TestExplorer` or `dotnet test` which both test using the `NUnit3TestAdapter`.
Most of our end-user also use the above or nunit-console which shares the engine with the adapter.

Why do we (still) use NUnitLite?

**Comment by OsirisTerje** [comment]
I believe @rprouse has to answer this one, but I do seem to remember something about not being dependent upon too much other stuff.  The dotnet test is dependent upon the adapter, which again depends on the engine, and if we change the public interface between the engine and the framework, like changing the xml format, or even, as Rob suggested, change away from the verbose xml to something more efficient, we might have trouble testing that the way we do testing now.  We could ofc test the framework using an older version of nunit, but right now we test using the framework itself both to execute the tests and as the SUT. 

PS:  About the "sharing" of the engine between adapter and console, they sort of do, but they are our of sync now wrt to versions.  Adapter is "stuck" on 3.15.4, whereas console is on 3.16.3 and main branch is on 4.0.........  and there is no team lead there.....

**Comment by rprouse** [comment]
We used to use the console to run all the tests but as we started adding NET Core targets we weren't able to test them using the console. The early dotnet test also didn't work well. Another advantage was easy debugging of the entire framework instead of just the tests.

NUnitLite is small and I like being able to create self running tests. I was thinking of including it directly in NUnit and dropping the nuget package.
